comment_spell_check/utils/create_checker.py

```diff
@@ -22,13 +24,44 @@ def create_checker(dict_list: list[str] = None) -> spellchecker.SpellChecker:
     english_dict = str(lib_path) + "/resources/en.json.gz"
     logger.info("Loading English dictionary from: %s", english_dict)
     checker.word_frequency.load_dictionary(english_dict)
+    logger.info("# of words: %d", checker.word_frequency.unique_words)
 
     # load the additional dictionaries
     if not isinstance(dict_list, list):
         return checker
     if len(dict_list) > 0:
         for d in dict_list:
-            logger.info("Loading additional dictionary from: %s", d)
-            checker.word_frequency.load_text_file(d)
+            if isinstance(d, pathlib.PosixPath):
+                # local file path
+                try:
+                    checker.word_frequency.load_text_file(d)
+                    logger.info("Loading dictionary: %s", d)
+                except IsADirectoryError:
+                    # if a directory is provided, load all text files in it
+                    for file in d.glob("*.txt"):
+                        try:
+                            checker.word_frequency.load_text_file(file)
+                            logger.info("Loading dictionary: %s", file)
+                        except FileNotFoundError:
+                            logger.error("File not found: %s", file)
+                            continue
+            else:
+                # load dictionary from URL
+                try:
+                    response = requests.get(d)
+                    response.raise_for_status()
+                    checker.word_frequency.load_text(response.text)
+                    logger.info("Loading dictionary URL: %s", d)
+                except requests.exceptions.MissingSchema:
+                    # URL didn't work so assume it's a local file path
+                    try:
+                        checker.word_frequency.load_text_file(d)
+                        logger.info("Loading dictionary: %s", d)
+                    except FileNotFoundError:
+                        logger.error("File not found: %s", d)
+                        continue
+                except requests.exceptions.RequestException as e:
```

Comment:
There is a lot less exception handing stuff in the example I like previously.  The complexity or depth of this code seems more than it needs to be.